File: codeforces/contests/Codeforces-Round-1014-Div-2/C.py
def solve():
    n = int(input())
    a = list(map(int, input().split()))
    
    odd_count = sum(1 for x in a if x % 2 != 0)
    
    if odd_count % 2 == 0:
        print(max(a))
    else:
        print(max(a) + 1 if max(a) == sum(a) // n and sum(a)%n !=0 else max(a))

t = int(input())
for _ in range(t):
    solve()

File: codeforces/contests/Codeforces-Round-1014-Div-2/C.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int t;
    if (!(cin >> t)) return 1;
    
    while (t--) {
        int n;
        if (!(cin >> n)) return 1;
        
        long long sum = 0;
        int oddCount = 0;
        int maxVal = 0;
        vector<int> a(n);
        
        for (int i = 0; i < n; i++) {
            cin >> a[i];
            sum += a[i];
            if (a[i] % 2 != 0)
                oddCount++;
            if (a[i] > maxVal)
                maxVal = a[i];
        }
        
        if (oddCount == n || oddCount == 0) {
            cout << maxVal << "\n";
        } else {
            long long ans = sum - (oddCount - 1);
            cout << ans << "\n";
        }
    }
    return 0;
}
